File: k8sbase/secrets.go
package k8sbase

import (
	"github.com/blesswinsamuel/infra-base/k8sapp"
	"github.com/blesswinsamuel/infra-base/packager"
)

type SecretsProps struct {
	ExternalSecrets    ExternalSecretsProps    `json:"externalSecrets"`
	DockerCreds        SecretsDockerCredsProps `json:"dockerCreds"`
	ClusterSecretStore ClusterSecretStoreProps `json:"clusterSecretStore"`
}

func NewSecrets(scope packager.Construct, props SecretsProps) packager.Construct {
	defer logModuleTiming("secrets")()

	chart := k8sapp.NewNamespaceChart(scope, "secrets")

	NewExternalSecrets(chart, props.ExternalSecrets)
	NewSecretsDockerCreds(chart, props.DockerCreds)
	NewClusterSecretStore(chart, props.ClusterSecretStore)
	return chart
}

File: k8sbase/secrets.ts
import { Construct } from "constructs";
// import { ClusterSecretStoreProps } from "./cluster-secret-store";
// import { ExternalSecretsProps } from "./external-secrets";
import { KubeNamespace } from "./imports/k8s.ts";
import {
  ExternalSecrets,
  ExternalSecretsProps,
} from "./secrets-external-secrets.ts";
// import { SecretsDockerCredsProps } from "./secrets-docker-creds";

export interface SecretsProps {
  readonly externalSecrets: ExternalSecretsProps;
  //   readonly dockerCreds: SecretsDockerCredsProps;
  //   readonly clusterSecretStore: ClusterSecretStoreProps;
}

export class Secrets extends Construct {
  constructor(scope: Construct, id: string, props: SecretsProps) {
    super(scope, id);

    new KubeNamespace(this, "secrets");

    new ExternalSecrets(this, "externalSecrets", props.externalSecrets);

    // new SecretsDockerCreds(this, "dockerCreds", props.dockerCreds);
    // new ClusterSecretStore(
    //   this,
    //   "clusterSecretStore",
    //   props.clusterSecretStore,
    // );
  }
}
